signin.js

```diff
@@ -1,90 +1,55 @@
-// 테스트 계정 설정
-const testCase = {
-  email: 'test@codeit.com',
-  password: 'codeit101',
-}
-
-// 이메일 형식 확인하는 정규화
-const emailForm = /^[0-9a-zA-Z]([-_\.]?[0-9a-zA-Z])*@[0-9a-zA-Z]([-_\.]?[0-9a-zA-Z])*\.[a-zA-Z]{2,3}$/i
+import { accountInfo, emailForm, passwordFrom } from "./javascript/account.js"
+import { inputFunction } from "./javascript/signFunction.js"
+import { passwordToggle } from "./javascript/passwordToggle.js"
 
 // 
 const inputEmail = document.querySelector('.email-input')
 const inputPassword = document.querySelector('.password-input')
-const loginButton = document.querySelector('.sign-btn')
-const togglePassword = document.querySelector('.eye-btn')
-
-// email foucus, blur 함수
-function emailFocus(e) {
-  inputEmail.classList.remove('input-error')
-  e.target.parentElement.lastElementChild.textContent = ''
-}
+const signButton = document.querySelector('.sign-btn')
 
-function emailBlur(e) {
-  if (e.target.value === '') {
-    e.target.classList.add('input-error')
-    e.target.parentElement.lastElementChild.textContent = '이메일을 입력해 주세요.'
-  }
-  else if (!e.target.value.match(emailForm)) {
-    e.target.classList.add('input-error')
-    e.target.parentElement.lastElementChild.textContent = '올바른 이메일 주소가 아닙니다.'
-  }
-}
-
-inputEmail.addEventListener('focus', emailFocus)
-inputEmail.addEventListener('blur', emailBlur)
+// 로그인 버튼, 경우에 따른 케이스 적용
+const submitButton = (e) => {
+  e.preventDefault()
 
-// password foucus, blur 함수
-function passwordFocus(e) {
-  inputPassword.classList.remove('input-error')
-  e.target.parentElement.parentElement.lastElementChild.textContent = ''
-}
+  const inputEmailValue = inputEmail.value
+  const inputPasswordValue = inputPassword.value
+  const matchingAccountEmail = accountInfo.find((item) => item.email === inputEmailValue)
+  const matchingAccountPassword = accountInfo.find((item) => item.password === inputPasswordValue)
+  
 
-function passwordBlur(e) {
-  if (e.target.value === '') {
-    e.target.classList.add('input-error')
-    e.target.parentElement.parentElement.lastElementChild.textContent = '비밀번호를 입력해 주세요.'
+  const inputErrorHandler = (inputType, errorMessage, type) => {
+    if (type === 'email') {
+      inputType.classList.add('input-error')
+      inputType.parentElement.lastElementChild.textContent = errorMessage
+    }
+    else {
+      inputType.classList.add('input-error')
+      inputType.parentElement.parentElement.lastElementChild.textContent = errorMessage
+    }
   }
-}
-
-inputPassword.addEventListener('focus', passwordFocus)
-inputPassword.addEventListener('blur', passwordBlur)
-
 
-// 로그인 버튼, 경우에 따른 케이스 적용
-function submitButton(e) {
-  e.preventDefault()
-  if (inputEmail.value === testCase.email && inputPassword.value === testCase.password) {
+  if (inputEmailValue === matchingAccountEmail?.email && inputPasswordValue === matchingAccountPassword?.password) {
     window.location.replace('folder.html')
```

Comment:
replace method를 사용하게 되면 뒤로가기 버튼이 동작하지 않습니다.
다른 방법으로 이동시키는 걸 추천드립니다.

```suggestion
    window.location.assign('folder.html')
```

Reply:
* mdn location replace 문서
https://developer.mozilla.org/en-US/docs/Web/API/Location/replace

Reply:
네 알겠습니다!!